pyfar/dsp/filter/audiofilter.py

```diff
@@ -349,6 +480,110 @@ def high_shelve_cascade(
+    frequency_type : string
+        Defines how `frequency` is used
+
+        ``'upper'``
+            `frequency` gives the upper characteristic frequency. In this case
+            the lower characteristic frequency is given by
+            ``2**bandwidth / frequency``
+        ``'lower'``
+            `frequency` gives the lower characteristic frequency. In this case
+            the upper characteristic frequency is given by
+            ``2**bandwidth * frequency``
+    gain : number
+        The filter gain in dB. The default is ``None``, which calculates the
+        gain from the `slope` and `bandwidth` (must be given if `gain` is
+        ``None``).
+    slope : number
+        Filter slope in dB per octave, with positive values denoting a rising
+        filter slope and negative values denoting a falling filter slope. The
+        default is ``None``, which calculates the slope from the `gain` and
+        `bandwidth` (must be given if `slope` is ``None``).
+    bandwidth : number
+        The bandwidth of the filter in octaves. The default is ``None``, which
+        calculates the bandwidth from `gain` and `slope` (must be given if
+        `bandwidth` is ``None``).
+    N : int
+        Number of shelf filters that are cascaded. The default is ``None``,
+        which calculated the minimum ``N`` that is required to satisfy Eq. (11)
+        in Schultz et al. 2020, i.e., the minimum ``N`` that is required for
+        a good approximation of the ideal filter response.
+    sampling_rate : None, number
+        The sampling rate in Hz. Only required if signal is ``None``. The
+        default is ``None``.
+
+    Returns
+    -------
+    signal : :py:class:`~pyfar.classes.audio.Signal`, :py:class:`~pyfar.classes.filter.FilterSOS`
+        The filtered signal (returned if ``sampling_rate = None``) or the
+        Filter object (returned if ``signal = None``).
+    N : int
+        The number of shelf filters that were cascaded
+    ideal : :py:class:`~pyfar.classes.audio.FrequencyData`
+        The ideal, piece-wise magnitude response of the filter
+
+    References
+    ----------
+    .. [#] F. Schultz, N. Hahn, and S. Spors, “Shelving Filter Cascade with
+           Adjustable Transition Slope and Bandwidth,” in 148th AES Convention
+           (Vienna, Austria, 2020).
+
+    Examples
+    --------
+
+    Generate a filter with a bandwith of 4 octaves and a gain of -60 dB and
+    compare it to the piece-wise constant idealized magnitude response.
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> import matplotlib.pyplot as plt
+        >>>
+        >>> impulse = pf.signals.impulse(40e3, sampling_rate=40000)
+        >>> impulse, N, ideal = pf.dsp.filter.high_shelf_cascade(
+        >>>     impulse, 250, "lower", -60, None, 4)
+        >>>
+        >>> pf.plot.freq(ideal, c='k', ls='--', label="ideal")
+        >>> pf.plot.freq(impulse, label="actual")
+        >>> plt.legend()
+    """
```

Comment:
```suggestion
    """  # noqa: E501
```
we always handle it in this way, as we know there is no other way